drivers/hmis/app/models/hmis/ce/referral_enroller.rb

```diff
@@ -55,11 +50,18 @@ def create_enrollment(message)
 
     # this is not actually an accessor method, even though RuboCop thinks it is
     def set_move_in_date(message) # rubocop:disable Naming/AccessorMethodName
-      project = referral.target_project
-      raise 'access denied' unless message.user.can_edit_enrollments_for?(project)
+      # No need to check permission for editing the enrollment.
+      # If the user can complete this step, they can do its side effects, even if they don't have direct permission
+
+      # Find the form item corresponding to move-in date
+      form_definition = message.step.form_definition # The form definition most recently used to complete the step
+      move_in_date_item = form_definition.link_id_item_hash.values.find do |item|
+        item['mapping']['field_name'] == 'moveInDate' && item['mapping']['record_type'] == 'ENROLLMENT'
```

Comment:
Since these are OpenStructs we can access like so
```suggestion
        item.mapping&.field_name == 'moveInDate' && item.mapping&.record_type == 'ENROLLMENT'
```